doc/plugins.rst

```diff
@@ -5,74 +5,175 @@ tox plugins
 
 .. versionadded:: 2.0
 
-With tox-2.0 a few aspects of tox running can be experimentally modified
-by writing hook functions.  The list of of available hook function is
-to grow over time on a per-need basis.
+A growing number of hooks make tox modifiable in different phases of execution by writing plugins.
 
+tox - like `pytest`_ and `devpi`_ - uses `pluggy`_ to provide an extension mechanism for pip-installable internal or devpi/PyPi-published plugins.
 
-writing a setuptools entrypoints plugin
----------------------------------------
+Using plugins
+-------------
 
-If you have a ``tox_MYPLUGIN.py`` module you could use the following
-rough ``setup.py`` to make it into a package which you can upload to the
-Python packaging index:
+To start using a plugin you need to install it in the same environment where the tox host
+is installed.
 
-.. code-block:: python
+e.g.:
 
-    # content of setup.py
-    from setuptools import setup
+.. code-block:: shell
+
+    $ pip install tox-travis
+
+You can search for available plugins on PyPi by typing ``pip search tox`` and filter for
+packages that are prefixed `tox-` or contain the "plugin" in the description.
+You will get some output similar to this::
+
+    tox-pipenv (1.4.1)                   - A pipenv plugin for tox
```

Comment:
a bit long, would say 5 entries as enough for example

Reply:
But shows that there are quite a few already :) In the long run it would be better to have the example there short and have a proper list with existing plugins. For now I'd say let's leave it like this.

Reply:
Can't we just link to https://pypi.org/search/?q=tox-? I know has a few false positives but still.